app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt

```diff
@@ -228,4 +249,67 @@ class ExposureNotificationRepositoryImpl(
                 }
         }
     }
+
+    override suspend fun registerAppForExposureNotificationsNow() {
+        suspendCancellableCoroutine { continuation: CancellableContinuation<Void> ->
+            exposureNotificationClient.start()
+                .addOnSuccessListener {
+                    continuation.resume(it)
+                }
+                .addOnFailureListener {
+                    continuation.cancel(it)
+                }
+        }
+    }
+
+    override suspend fun processBatchDiagnosisKeys(archives: List<File>, token: String): String {
+
+        //TODO get values from configuration
+        val configuration = ExposureConfiguration.ExposureConfigurationBuilder()
+            .setDurationAtAttenuationThresholds(50, 60)
+            .setMinimumRiskScore(1)
+            .setDaysSinceLastExposureScores(1,2,3,4,5,6,7,8)
+            .setDurationScores(1,2,3,4,5,6,7,8)
+            .setAttenuationScores(1,2,3,4,5,6,7,8)
+            .setDaysSinceLastExposureWeight(100)
+            .setTransmissionRiskWeight(100)
+            .build()
+
+        return suspendCancellableCoroutine { continuation ->
+            exposureNotificationClient.provideDiagnosisKeys(archives, configuration, token)
+                .addOnCompleteListener{
+                    if (it.isSuccessful) {
+                        continuation.resume(token)
```

Comment:
Do I understand well, you just fulfilling the exposure client with the data? And as a result, we can get an event in `ExposureNotificationBroadcastReceiver`? If so, this should be described more in Kdoc.

Reply:
@falkorichter

Reply:
Do not return the token